tests/test_visualizer.py

```diff
@@ -292,3 +293,47 @@ def test_remove() -> None:
     assert viser_model._traces == [], "remove did not clear _traces"
     # Note: geometries are removed via their remove() method; further checking
     # would require inspection of ViserServer's scene state.
+
+
+@pytest.mark.parametrize(
+    "xml, expected_bodies, expected_geoms",
+    [
+        (
+            # all unnamed
+            """
+        <mujoco>
+          <worldbody>
+            <body><geom type="sphere" size=".1"/></body>
+            <body><geom type="sphere" size=".1"/></body>
+          </worldbody>
+        </mujoco>
+        """,
+            ["JUDO_BODY_0", "JUDO_BODY_1"],
+            ["JUDO_GEOM_0", "JUDO_GEOM_1"],
+        ),
+        (
+            # mixed names
+            """
+        <mujoco>
+          <worldbody>
+            <body name="body_0"><geom type="sphere" size=".1"/></body>
+            <body><geom name="geom_0" type="sphere" size=".1"/></body>
+            <body><geom type="sphere" size=".1"/></body>
+          </worldbody>
+        </mujoco>
+        """,
+            ["body_0", "JUDO_BODY_0", "JUDO_BODY_1"],
+            ["JUDO_GEOM_0", "geom_0", "JUDO_GEOM_1"],
+        ),
+    ],
+)
```

Comment:
this works, but is a bit clunky to read - i would consider defining two pytest fixtures above and just manually referencing and testing them both below